python_modules/dagster/dagster/_core/definitions/multi_dimensional_partitions.py

```diff
@@ -0,0 +1,52 @@
+from typing import List, Mapping, NamedTuple
+
+import dagster._check as check
+
+
+class PartitionDimensionKey(
+    NamedTuple("_PartitionDimensionKey", [("dimension_name", str), ("partition_key", str)])
+):
+    """
+    Representation of a single dimension of a multi-dimensional partition key.
+    """
+
+    def __new__(cls, dimension_name: str, partition_key: str):
+        return super(PartitionDimensionKey, cls).__new__(
+            cls,
+            dimension_name=check.str_param(dimension_name, "dimension_name"),
+            partition_key=check.str_param(partition_key, "partition_key"),
+        )
+
+
+class MultiPartitionKey(str):
```

Comment:
Do we end up serializing this? How does that end up getting handled?

Reply:
yes, this ends up getting serialized. So we directly serialize the stringified partition key e.g. `US|2022-10-10` into storage. This won't cause breakages for users though--we read from the tags to learn whether a string is a multi-partition key or not

Reply:
Got it - makes sense